### PR TITLE
MM-44953: Use plugin API for ensure bot

### PR DESCRIPTION
**Pull request by agnivade**
We use the refactored plugin API for ensure bot user.

https://mattermost.atlassian.net/browse/MM-44953

### CODE REVIEW

**Comment by agnivade** [comment]
It's pointing to the PR branch now. To be changed to master once the server PR is merged.

**Review by isacikgoz — APPROVED** [review]
Good to go after server PR merged.

**Pull request merged**
